aexpect/ops.py

```diff
@@ -0,0 +1,439 @@
+
+    :param session: session to run the command on
+    :type session: ShellSession
+    :param str path: path to the regular file to check
+    :returns: whether the regular file exists
+    :rtype: bool
+    """
+    return test(session, path, flags="-f")
+
+
+def is_nonzero_size_file(session, path):
+    """
+    Check if a regular file exists on a given remote host.
+
+    :param session: session to run the command on
+    :type session: ShellSession
+    :param str path: path to the regular file to check
+    :returns: whether the regular file exists and is nonzero size
+    :rtype: bool
+    """
+    return test(session, path, flags="-s")
+
+
+###############################################################################
+# file ops
+###############################################################################
+
+
+def hash_file(session, filename, method='md5'):
+    """
+    Calculate hash of given file from a session.
+
+    :param session: session to run the command on
+    :type session: ShellSession
+    :param str filename: full path of file that should be hashed
+    :param str method: method for hashing
+    :returns: hash of file, a hex number
+    :rtype: str
+    :raises: :py:class:`RuntimeError` if hash command failed (e.g. file not
+             found) or resulting output does not have expected format
+    """
+    cmd = f"{method}sum"
+    expect_len = 32
+
+    # run cmd on shell
+    status, output = session.cmd_status_output(cmd + ' ' + quote(filename))
+    if output:
+        output = output.strip()
+    if status != 0 or not output:
+        raise RuntimeError(f'Could not hash {filename} using {cmd}: {output}')
+
+    # parse output
+    hash_str = output.split(maxsplit=1)[0].lower()
+
+    # check length and all chars are hex
+    if expect_len and len(hash_str) != expect_len:
+        raise RuntimeError(f'Resulting hash string has unexpected length '
+                           f'{len(hash_str)}: {hash_str}')
+    if hash_str.strip('0123456789abcdef'):
+        raise RuntimeError('Resulting hash string has unexpected characters: '
+                           + hash_str)
+    return hash_str
+
+
+def extract_tarball(session, tarball, target_dir):
+    """
+    Extract tarball to given dir from a session.
+
+    :param session: session to run the command on
+    :type session: ShellSession
+    :param str tarball: full path of tar file that should be extracted
+    :param str target_dir: name of directory where tarball should be extracted
+    :raises: :py:class:`RuntimeError` if tar command returned non-null
+    """
+    cmd = f'tar -C {quote(target_dir)} --strip-components=1 -xapf {quote(tarball)}'
+    status, output = session.cmd_status_output(cmd)
+    if output:
+        output = output.strip()
+    if status != 0:
+        raise RuntimeError(f'Failed to extract {tarball} to {target_dir}: {output}')
+
+
+def ls(session, path, is_glob=False, flags="-1UNq"):  # pylint: disable=C0103
```

Comment:
I was wondering about a proper name here too and thought about using some of names indicating quotation but my problem with this was that it doesn't indicate immediately the rather widely spread glob expansion usage. OTOH a user might expect path expansion by default to begin but then this would contradict with the quote usage for safety and space containing paths handling. So let's stick to `quote_path` and generalize it to one or two other file operations but stay at that for the time being.

About the args order I would definitely prefer it the `flags` that will likely end up being shared by all functions's signature will be the last one with more function-specific arguments coming beforehand.